recipes/npcap/all/test_package/conanfile.py

```diff
@@ -0,0 +1,49 @@
+from conan import ConanFile
+from conan.tools import files
+from conan.tools.build import can_run
+from conan.tools.cmake import cmake_layout, CMake
+from io import StringIO
+import os
+
+
+# It will become the standard on Conan 2.x
+class TestPackageConan(ConanFile):
+    settings = "os", "arch", "compiler", "build_type"
+    generators = "CMakeDeps", "CMakeToolchain", "VirtualRunEnv"
+    test_type = "explicit"
+    # Needed to make sure libpcap gets compiled
+    build_policy = "missing"
+
+    def requirements(self):
+        self.requires(self.tested_reference_str)
+        if can_run(self):
+            self.requires("libpcap/1.10.1")
+
+    def configure(self):
+        if can_run(self):
+            self.options["libpcap"].shared = True
+
+    def layout(self):
+        cmake_layout(self)
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def test(self):
+        if can_run(self):
+            bindir = self.cpp.build.bindirs[0]
+            # Use libpcap DLL as a replacement for npcap DLL
+            # It will not provide all the functions
+            # but it will cover enough to check that what we complied is correct
+            files.rm(self, "wpcap.dll", bindir)
+            files.copy(self, "pcap.dll", src=self.deps_cpp_info['libpcap'].bin_paths[0], dst=bindir)
+            files.rename(self, os.path.join(bindir, "pcap.dll"), os.path.join(bindir, "wpcap.dll"))
```

Comment:
An alternative to the copy/rename is to use [/DELAYLOAD](https://docs.microsoft.com/en-us/cpp/build/reference/linker-support-for-delay-loaded-dlls?view=vs-2019) + `__pfnDliFailureHook2` as described in https://stackoverflow.com/a/55497880/1438522 but that only works with MSVC.

Not sure if you prefer the conan/python or C way of handling different DLL name.

Reply:
No, the test is just to validate the package, workarounds should not be required. If that method returns a string with an error, it's not a problem. It should validate if we can link the library, if the cmake target is correct, if headers are installed ...

Reply:
So just compiling test_package without actually running it is enough?

Reply:
Runtime linkage requires the presence of a 'wpcap.dll' this is not included in conan package since it must use the system installed one instead (to match the driver installed on the system)

since installing third-party packages is not allowed in conan ci, I need to provide the test with an alternative dll (libpcap for example). otherwise the test program won't even start due to the missing dll.

Reply:
Sorry, it's worst than I thought. I had in mind only asking project's version to avoid any .dll. Well, so we will need to add winpcap package and add it as pcap's dependency. I could take a look during the weekend.

Reply:
winpcap itself is no longer maintained by its authors... are you sure you want to spend time trying to get it to compile?

From https://www.winpcap.org/default.htm
> WinPcap, though still available for download (v4.1.3), has not seen an upgrade in many years and there are no road map/future plans to update the technology. While community support may persist, technical oversight by Riverbed staff, responses to questions posed by Riverbed resources, and bug reporting are no longer available.

> Gordon Lyon, Nmap project founder, has created Npcap, a packet capture library for Windows, that includes WinPcap compatibility and may be a suitable replacement for WinPcap and WinPcap Pro. Information can be found at https://nmap.org/npcap/. 

If you just need to get a "wpcap.dll" for winpcap you can simply extract it from the installer 
```
wget https://www.winpcap.org/install/bin/WinPcap_4_1_3.exe
7z -aou e "WinPcap_4_1_3.exe"
```
Lib and headers could be found in the developer pack https://www.winpcap.org/devel.htm

Reply:
Okay, so let's use your workaround related to rename the .dll file which should work on runtime.

Reply:
@uilianries anything else or this PR looks good to you?